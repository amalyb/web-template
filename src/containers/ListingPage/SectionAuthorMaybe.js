import React from 'react';
import { FormattedMessage } from '../../util/reactIntl';
import { UserCard, Modal } from '../../components';
import InquiryForm from './InquiryForm/InquiryForm';

import css from './ListingPage.module.css';

const SectionAuthorMaybe = props => {
  const {
    title,
    listing,
    authorDisplayName,
    onContactUser,
    isInquiryModalOpen,
    onCloseInquiryModal,
    sendInquiryError,
    sendInquiryInProgress,
    onSubmitInquiry,
    currentUser,
    onManageDisableScrolling,
  } = props;

  if (!listing.author) {
    return null;
  }

  return (
    <div id="author" className={css.sectionAuthor}>
      <h2 className={css.aboutProviderTitle}>
        <FormattedMessage id="ListingPage.aboutProviderTitle" />
      </h2>
      <UserCard user={listing.author} currentUser={currentUser} onContactUser={onContactUser} />
      <Modal
        id="ListingPage.inquiry"
        contentClassName={css.inquiryModalContent}
        isOpen={isInquiryModalOpen}
        onClose={onCloseInquiryModal}
        usePortal
        onManageDisableScrolling={onManageDisableScrolling}
      >
        <InquiryForm
          className={css.inquiryForm}
          submitButtonWrapperClassName={css.inquirySubmitButtonWrapper}
          listingTitle={title}
          authorDisplayName={authorDisplayName}
          sendInquiryError={sendInquiryError}
          onSubmit={onSubmitInquiry}
          inProgress={sendInquiryInProgress}
        />
      </Modal>
    </div>
  );
};

export default SectionAuthorMaybe;
