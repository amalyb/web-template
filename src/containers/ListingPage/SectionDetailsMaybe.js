import React from 'react';
import { FormattedMessage } from '../../util/reactIntl';
import { PropertyGroup } from '../../components';

import css from './ListingPage.module.css';

const SectionDetailsMaybe = props => {
  const { publicData, metadata = {}, customConfig, intl } = props;
  const { listingExtendedData } = customConfig || {};

  if (!publicData || !customConfig) {
    return null;
  }

  const pickExtendedData = (filteredConfigs, config) => {
    const { key, schemaType, schemaOptions, listingPageConfig = {} } = config;
    const { isDetail, label } = listingPageConfig;
    const publicDataValue = publicData[key];
    const metadataValue = metadata[key];
    const value = publicDataValue || metadataValue;

    if (isDetail && typeof value !== 'undefined') {
      const findSelectedOption = enumValue =>
        schemaOptions.find(o => enumValue === `${o}`.toLowerCase().replace(/\s/g, '_'));
      const getBooleanMessage = value =>
        value
          ? intl.formatMessage({ id: 'SearchPage.detailYes' })
          : intl.formatMessage({ id: 'SearchPage.detailNo' });
      return schemaType === 'enum'
        ? filteredConfigs.concat({ key, value: findSelectedOption(value), label })
        : schemaType === 'boolean'
        ? filteredConfigs.concat({ key, value: getBooleanMessage(value), label })
        : schemaType === 'long'
        ? filteredConfigs.concat({ key, value, label })
        : null;
    }
    return filteredConfigs;
  };

  const existingExtendedData = listingExtendedData.reduce(pickExtendedData, []);

  return existingExtendedData ? (
    <div className={css.sectionDetails}>
      <h2 className={css.detailsTitle}>
        <FormattedMessage id="ListingPage.detailsTitle" />
      </h2>
      <ul className={css.details}>
        {existingExtendedData.map(detail => (
          <li key={detail.key} className={css.detailsRow}>
            <span className={css.detailLabel}>{detail.label}</span>
            <span>{detail.value}</span>
          </li>
        ))}
      </ul>
    </div>
  ) : null;
};

export default SectionDetailsMaybe;
