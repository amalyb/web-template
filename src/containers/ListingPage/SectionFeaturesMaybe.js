import React from 'react';
import { PropertyGroup } from '../../components';

import css from './ListingPage.module.css';

const SectionFeaturesMaybe = props => {
  const { heading, options, selectedOptions } = props;
  if (!heading || !options || !selectedOptions) {
    return null;
  }

  return (
    <div className={css.sectionFeatures}>
      <h2 className={css.featuresTitle}>{heading}</h2>
      <PropertyGroup
        id="ListingPage.amenities"
        options={options}
        selectedOptions={selectedOptions}
        twoColumns={options.length > 5}
      />
    </div>
  );
};

export default SectionFeaturesMaybe;
