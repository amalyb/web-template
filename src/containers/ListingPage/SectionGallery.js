import React from 'react';
import config from '../../config';
import ListingImageGallery from './ListingImageGallery/ListingImageGallery';

import css from './ListingPage.module.css';

const SectionGallery = props => {
  const { listing } = props;
  const images = listing.images;
  const { variantPrefix } = config.listing;
  const variants = [variantPrefix, `${variantPrefix}-2x`, `${variantPrefix}-4x`];
  return (
    <ListingImageGallery
      rootClassName={css.productGallery}
      images={images}
      imageVariants={variants}
    />
  );
};

export default SectionGallery;
