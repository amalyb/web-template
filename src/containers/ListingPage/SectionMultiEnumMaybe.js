import React from 'react';
import { PropertyGroup } from '../../components';

import css from './ListingPage.module.css';

const SectionMultiEnumMaybe = props => {
  const { heading, options, selectedOptions } = props;
  if (!heading || !options || !selectedOptions) {
    return null;
  }

  return (
    <div className={css.sectionMultiEnum}>
      <h2 className={css.multiEnumTitle}>{heading}</h2>
      <PropertyGroup
        id="ListingPage.amenities"
        options={options}
        selectedOptions={selectedOptions}
        twoColumns={options.length > 5}
      />
    </div>
  );
};

export default SectionMultiEnumMaybe;
