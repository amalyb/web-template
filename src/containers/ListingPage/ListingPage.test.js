import React from 'react';
import { shallow } from 'enzyme';

import { FormattedMessage } from '../../util/reactIntl';
import { types as sdkTypes } from '../../util/sdkLoader';
import {
  createUser,
  createCurrentUser,
  createListing,
  createOwnListing,
  fakeIntl,
} from '../../util/test-data';
import { storableError } from '../../util/errors';
import {
  renderShallow,
  getRouteConfiguration,
  getDefaultConfiguration,
} from '../../util/test-helpers';
import {
  LISTING_STATE_PENDING_APPROVAL,
  LISTING_STATE_PUBLISHED,
  LISTING_STATE_CLOSED,
} from '../../util/types';
import { addMarketplaceEntities } from '../../ducks/marketplaceData.duck';
import { showListingRequest, showListingError, showListing } from './ListingPage.duck';

import { ListingPageComponent } from './ListingPageHeroImage';
import ActionBarMaybe from './ActionBarMaybe';

const { UUID } = sdkTypes;
const noop = () => null;

const listingConfig = {
  listingExtendedData: [
    {
      key: 'category',
      scope: 'public',
      includeForListingTypes: ['sell-bicycles'],
      schemaType: 'enum',
      schemaOptions: [{ option: 'cat_1', label: 'Cat 1' }, { option: 'cat_2', label: 'Cat 2' }],
      indexForSearch: true,
      listingPageConfig: {
        label: 'Category',
        isDetail: true,
      },
    },
    {
      key: 'amenities',
      scope: 'public',
      includeForListingTypes: [
        'rent-bicycles-daily',
        'rent-bicycles-nightly',
        'rent-bicycles-hourly',
      ],
      schemaType: 'multi-enum',
      schemaOptions: [
        { option: 'feat_1', label: 'Feat 1' },
        { option: 'feat_2', label: 'Feat 2' },
        { option: 'feat_3', label: 'Feat 3' },
      ],
      indexForSearch: true,
      listingPageConfig: {
        label: 'Amenities',
        searchMode: 'has_all',
        group: 'secondary',
      },
    },
  ],
};
describe('ListingPage', () => {
  it('matches snapshot', () => {
    const currentUser = createCurrentUser('user-2');
    const id = 'listing1';
    const slug = 'listing1-title';
    const listing1 = createListing(id, {}, { author: createUser('user-1') });
    const listing1Own = createOwnListing(id, {}, { author: createCurrentUser('user-1') });
    const getListing = () => listing1;
    const getOwnListing = () => listing1Own;

    const props = {
      location: {
        pathname: `/l/${slug}/${id}`,
        search: '',
        hash: '',
      },
      history: {
        push: () => console.log('HistoryPush called'),
      },
      params: { id, slug },
      currentUser,
      getListing,
      getOwnListing,
      intl: fakeIntl,
      authInProgress: false,
      currentUserHasListings: false,
      isAuthenticated: false,
      onLogout: noop,
      onLoadListing: noop,
      onManageDisableScrolling: noop,
      scrollingDisabled: false,
      callSetInitialValues: noop,
      sendVerificationEmailInProgress: false,
      onResendVerificationEmail: noop,
      onInitializeCardPaymentData: noop,
      sendInquiryInProgress: false,
      onSendInquiry: noop,
      listingConfig,
      fetchLineItemsInProgress: false,
      onFetchTransactionLineItems: () => null,
      onFetchTimeSlots: () => null,
      config: getDefaultConfiguration(),
      routeConfiguration: getRouteConfiguration(),
    };

    const tree = renderShallow(<ListingPageComponent {...props} />);
    expect(tree).toMatchSnapshot();
  });

  describe('Duck', () => {
    const listingExtendedData = [];
    const config = {
      layout: {
        listingImage: {
          aspectWidth: 400,
          aspectHeight: 400,
          variantPrefix: 'listing-card',
        },
      },
      listing: {
        listingExtendedData,
      },
    };

    it('showListing() success', () => {
      const id = new UUID('00000000-0000-0000-0000-000000000000');
      const dispatch = jest.fn(action => action);
      const response = { status: 200 };
      const show = jest.fn(() => Promise.resolve(response));
      const sdk = { listings: { show }, currentUser: { show } };

      return showListing(id, config)(dispatch, null, sdk).then(data => {
        expect(data).toEqual(response);
        expect(show.mock.calls).toEqual([
          [
            expect.objectContaining({
              id,
              'imageVariant.listing-card': 'w:400;h:400;fit:crop',
              'imageVariant.listing-card-2x': 'w:800;h:800;fit:crop',
              include: ['author', 'author.profileImage', 'images', 'currentStock'],
            }),
          ],
        ]);
        expect(dispatch.mock.calls).toEqual([
          [showListingRequest(id)],
          [expect.anything()], // fetchCurrentUser() call
          [addMarketplaceEntities(data, { listingExtendedData })],
        ]);
      });
    });

    it('showListing() error', () => {
      const id = new UUID('00000000-0000-0000-0000-000000000000');
      const dispatch = jest.fn(action => action);
      const error = new Error('fail');
      const show = jest.fn(() => Promise.reject(error));
      const sdk = { listings: { show } };

      // Calling sdk.listings.show is expected to fail now

      return showListing(id, config)(dispatch, null, sdk).then(data => {
        expect(show.mock.calls).toEqual([
          [
            expect.objectContaining({
              id,
              'imageVariant.listing-card': 'w:400;h:400;fit:crop',
              'imageVariant.listing-card-2x': 'w:800;h:800;fit:crop',
              include: ['author', 'author.profileImage', 'images', 'currentStock'],
            }),
          ],
        ]);
        expect(dispatch.mock.calls).toEqual([
          [showListingRequest(id)],
          [expect.anything()], // fetchCurrentUser() call
          [showListingError(storableError(error))],
        ]);
      });
    });
  });

  describe('ActionBarMaybe', () => {
    it('shows users own listing status', () => {
      const listing = createListing('listing-published', {
        state: LISTING_STATE_PUBLISHED,
      });
      const actionBar = shallow(<ActionBarMaybe isOwnListing listing={listing} editParams={{}} />);
      const formattedMessages = actionBar.find(FormattedMessage);
      expect(formattedMessages.length).toEqual(2);
      expect(formattedMessages.at(0).props().id).toEqual('ListingPage.ownListing');
      expect(formattedMessages.at(1).props().id).toEqual('ListingPage.editListing');
    });
    it('shows users own pending listing status', () => {
      const listing = createListing('listing-published', {
        state: LISTING_STATE_PENDING_APPROVAL,
      });
      const actionBar = shallow(<ActionBarMaybe isOwnListing listing={listing} editParams={{}} />);
      const formattedMessages = actionBar.find(FormattedMessage);
      expect(formattedMessages.length).toEqual(2);
      expect(formattedMessages.at(0).props().id).toEqual('ListingPage.ownListingPendingApproval');
      expect(formattedMessages.at(1).props().id).toEqual('ListingPage.editListing');
    });
    it('shows users own closed listing status', () => {
      const listing = createListing('listing-closed', {
        state: LISTING_STATE_CLOSED,
      });
      const actionBar = shallow(<ActionBarMaybe isOwnListing listing={listing} editParams={{}} />);
      const formattedMessages = actionBar.find(FormattedMessage);
      expect(formattedMessages.length).toEqual(2);
      expect(formattedMessages.at(0).props().id).toEqual('ListingPage.ownClosedListing');
      expect(formattedMessages.at(1).props().id).toEqual('ListingPage.editListing');
    });
    it('shows closed listing status', () => {
      const listing = createListing('listing-closed', {
        state: LISTING_STATE_CLOSED,
      });
      const actionBar = shallow(
        <ActionBarMaybe isOwnListing={false} listing={listing} editParams={{}} />
      );
      const formattedMessages = actionBar.find(FormattedMessage);
      expect(formattedMessages.length).toEqual(1);
      expect(formattedMessages.at(0).props().id).toEqual('ListingPage.closedListing');
    });
    it("is missing if listing is not closed or user's own", () => {
      const listing = createListing('listing-published', {
        state: LISTING_STATE_PUBLISHED,
      });
      const actionBar = shallow(
        <ActionBarMaybe isOwnListing={false} listing={listing} editParams={{}} />
      );
      expect(actionBar.getElement()).toBeNull();
    });
  });
});
