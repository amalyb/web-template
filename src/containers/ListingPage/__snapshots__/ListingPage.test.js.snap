// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingPage matches snapshot 1`] = `
<Page
  author="user-1 display name"
  contentType="website"
  description="listing1 description"
  facebookImages={Array []}
  schema={
    Object {
      "@context": "http://schema.org",
      "@type": "Product",
      "description": "listing1 description",
      "image": Array [],
      "name": "ListingPage.schemaTitle",
      "offers": Object {
        "@type": "Offer",
        "availability": "https://schema.org/OutOfStock",
        "price": 55,
        "priceCurrency": "USD",
        "url": "http://localhost:3000/l/listing1-title/listing1",
      },
    }
  }
  scrollingDisabled={false}
  title="ListingPage.schemaTitle"
  twitterImages={Array []}
>
  <LayoutSingleColumn
    className="pageRoot"
    rootClassName={null}
  >
    <LayoutWrapperTopbar
      className={null}
      rootClassName={null}
    >
      <withRouter(Connect(TopbarContainerComponent)) />
    </LayoutWrapperTopbar>
    <LayoutWrapperMain
      className={null}
      rootClassName={null}
    >
      <div>
        <SectionHero
          editParams={
            Object {
              "id": "listing1",
              "slug": "listing1-title",
              "tab": "details",
              "type": "edit",
            }
          }
          handleViewPhotosClick={[Function]}
          imageCarouselOpen={false}
          isOwnListing={false}
          listing={
            Object {
              "attributes": Object {
                "deleted": false,
                "description": "listing1 description",
                "geolocation": LatLng {
                  "_sdkType": "LatLng",
                  "lat": 40,
                  "lng": 60,
                },
                "price": Money {
                  "_sdkType": "Money",
                  "amount": 5500,
                  "currency": "USD",
                },
                "publicData": Object {},
                "state": "published",
                "title": "listing1 title",
              },
              "author": Object {
                "attributes": Object {
                  "banned": false,
                  "deleted": false,
                  "profile": Object {
                    "abbreviatedName": "TT",
                    "displayName": "user-1 display name",
                  },
                },
                "id": UUID {
                  "_sdkType": "UUID",
                  "uuid": "user-1",
                },
                "type": "user",
              },
              "id": UUID {
                "_sdkType": "UUID",
                "uuid": "listing1",
              },
              "images": Array [],
              "type": "listing",
            }
          }
          onImageCarouselClose={[Function]}
          onManageDisableScrolling={[Function]}
          title="listing1 title"
        />
        <div
          className="contentWrapperForHeroLayout"
        >
          <div
            className="mainColumnForHeroLayout"
          >
            <div
              className="mobileHeading"
            >
              <h1
                className="orderPanelTitle"
              >
                <MemoizedFormattedMessage
                  id="ListingPage.orderTitle"
                  values={
                    Object {
                      "title": <span>
                        
                        listing1
                         
                        title
                        
                      </span>,
                    }
                  }
                />
              </h1>
            </div>
            <SectionTextMaybe
              showAsIngress={true}
              text="listing1 description"
            />
            <SectionDetailsMaybe
              intl={
                Object {
                  "formatDate": [Function],
                  "formatDateTimeRange": [Function],
                  "formatMessage": [Function],
                  "formatNumber": [Function],
                  "formatPlural": [Function],
                  "formatRelativeTime": [Function],
                  "formatTime": [Function],
                  "now": [Function],
                }
              }
              listingConfig={
                Object {
                  "listingExtendedData": Array [
                    Object {
                      "includeForTransactionTypes": Array [
                        "sell-bicycles",
                      ],
                      "indexForSearch": true,
                      "key": "category",
                      "listingPageConfig": Object {
                        "isDetail": true,
                        "label": "Category",
                      },
                      "schemaOptions": Array [
                        Object {
                          "label": "Cat 1",
                          "option": "cat_1",
                        },
                        Object {
                          "label": "Cat 2",
                          "option": "cat_2",
                        },
                      ],
                      "schemaType": "enum",
                      "scope": "public",
                    },
                    Object {
                      "includeForTransactionTypes": Array [
                        "rent-bicycles-daily",
                        "rent-bicycles-nightly",
                        "rent-bicycles-hourly",
                      ],
                      "indexForSearch": true,
                      "key": "amenities",
                      "listingPageConfig": Object {
                        "group": "secondary",
                        "label": "Amenities",
                        "searchMode": "has_all",
                      },
                      "schemaOptions": Array [
                        Object {
                          "label": "Feat 1",
                          "option": "feat_1",
                        },
                        Object {
                          "label": "Feat 2",
                          "option": "feat_2",
                        },
                        Object {
                          "label": "Feat 3",
                          "option": "feat_3",
                        },
                      ],
                      "schemaType": "multi-enum",
                      "scope": "public",
                    },
                  ],
                }
              }
              metadata={Object {}}
              publicData={Object {}}
            />
            <SectionMultiEnumMaybe
              heading="Amenities"
              key="amenities"
              options={
                Array [
                  Object {
                    "key": "feat_1",
                    "label": "Feat 1",
                  },
                  Object {
                    "key": "feat_2",
                    "label": "Feat 2",
                  },
                  Object {
                    "key": "feat_3",
                    "label": "Feat 3",
                  },
                ]
              }
            />
            <SectionMapMaybe
              className={null}
              geolocation={
                LatLng {
                  "_sdkType": "LatLng",
                  "lat": 40,
                  "lng": 60,
                }
              }
              listingId={
                UUID {
                  "_sdkType": "UUID",
                  "uuid": "listing1",
                }
              }
              mapsConfig={
                Object {
                  "fuzzy": Object {
                    "circleColor": "#c0392b",
                    "defaultZoomLevel": 13,
                    "enabled": false,
                    "offset": 500,
                  },
                  "googleMapsAPIKey": undefined,
                  "mapProvider": "MAPBOX",
                  "mapboxAccessToken": undefined,
                  "search": Object {
                    "currentLocationBoundsDistance": 1000,
                    "defaults": Array [],
                    "sortSearchByDistance": false,
                    "suggestCurrentLocation": true,
                  },
                }
              }
              publicData={Object {}}
              rootClassName={null}
            />
            <SectionReviews
              fetchReviewsError={null}
              reviews={Array []}
            />
            <SectionAuthorMaybe
              authorDisplayName="user-1 display name"
              currentUser={
                Object {
                  "attributes": Object {
                    "banned": false,
                    "deleted": false,
                    "email": "user-2@example.com",
                    "emailVerified": true,
                    "profile": Object {
                      "abbreviatedName": "user-2 abbreviated name",
                      "displayName": "user-2 display name",
                      "firstName": "user-2 first name",
                      "lastName": "user-2 last name",
                    },
                  },
                  "id": UUID {
                    "_sdkType": "UUID",
                    "uuid": "user-2",
                  },
                  "type": "currentUser",
                }
              }
              isEnquiryModalOpen={false}
              listing={
                Object {
                  "attributes": Object {
                    "deleted": false,
                    "description": "listing1 description",
                    "geolocation": LatLng {
                      "_sdkType": "LatLng",
                      "lat": 40,
                      "lng": 60,
                    },
                    "price": Money {
                      "_sdkType": "Money",
                      "amount": 5500,
                      "currency": "USD",
                    },
                    "publicData": Object {},
                    "state": "published",
                    "title": "listing1 title",
                  },
                  "author": Object {
                    "attributes": Object {
                      "banned": false,
                      "deleted": false,
                      "profile": Object {
                        "abbreviatedName": "TT",
                        "displayName": "user-1 display name",
                      },
                    },
                    "id": UUID {
                      "_sdkType": "UUID",
                      "uuid": "user-1",
                    },
                    "type": "user",
                  },
                  "id": UUID {
                    "_sdkType": "UUID",
                    "uuid": "listing1",
                  },
                  "images": Array [],
                  "type": "listing",
                }
              }
              onCloseEnquiryModal={[Function]}
              onContactUser={[Function]}
              onManageDisableScrolling={[Function]}
              onSubmitEnquiry={[Function]}
              sendEnquiryError={null}
              sendEnquiryInProgress={false}
              title="listing1 title"
            />
          </div>
          <div
            className="orderColumnForHeroLayout"
          >
            <withRouter(injectIntl(OrderPanel))
              author={
                Object {
                  "attributes": Object {
                    "banned": false,
                    "deleted": false,
                    "profile": Object {
                      "abbreviatedName": "TT",
                      "displayName": "user-1 display name",
                    },
                  },
                  "id": UUID {
                    "_sdkType": "UUID",
                    "uuid": "user-1",
                  },
                  "type": "user",
                }
              }
              authorLink={
                <NamedLink
                  className="authorNameLink"
                  name="ListingPage"
                  params={
                    Object {
                      "id": "listing1",
                      "slug": "listing1-title",
                    }
                  }
                  to={
                    Object {
                      "hash": "#author",
                    }
                  }
                >
                  user-1 display name
                </NamedLink>
              }
              className="orderPanel"
              dayCountAvailableForBooking={90}
              fetchLineItemsError={null}
              fetchLineItemsInProgress={false}
              isOwnListing={false}
              lineItems={null}
              listing={
                Object {
                  "attributes": Object {
                    "deleted": false,
                    "description": "listing1 description",
                    "geolocation": LatLng {
                      "_sdkType": "LatLng",
                      "lat": 40,
                      "lng": 60,
                    },
                    "price": Money {
                      "_sdkType": "Money",
                      "amount": 5500,
                      "currency": "USD",
                    },
                    "publicData": Object {},
                    "state": "published",
                    "title": "listing1 title",
                  },
                  "author": Object {
                    "attributes": Object {
                      "banned": false,
                      "deleted": false,
                      "profile": Object {
                        "abbreviatedName": "TT",
                        "displayName": "user-1 display name",
                      },
                    },
                    "id": UUID {
                      "_sdkType": "UUID",
                      "uuid": "user-1",
                    },
                    "type": "user",
                  },
                  "id": UUID {
                    "_sdkType": "UUID",
                    "uuid": "listing1",
                  },
                  "images": Array [],
                  "type": "listing",
                }
              }
              marketplaceCurrency="USD"
              marketplaceName="MarketplaceX"
              monthlyTimeSlots={null}
              onContactUser={[Function]}
              onFetchTimeSlots={[Function]}
              onFetchTransactionLineItems={[Function]}
              onManageDisableScrolling={[Function]}
              onSubmit={[Function]}
              title={
                <Memo(MemoizedFormattedMessage)
                  id="ListingPage.orderTitle"
                  values={
                    Object {
                      "title": <span>
                        
                        listing1
                         
                        title
                        
                      </span>,
                    }
                  }
                />
              }
              titleDesktop={
                <h1
                  className="orderPanelTitle"
                >
                  <Memo(MemoizedFormattedMessage)
                    id="ListingPage.orderTitle"
                    values={
                      Object {
                        "title": <span>
                          
                          listing1
                           
                          title
                          
                        </span>,
                      }
                    }
                  />
                </h1>
              }
            />
          </div>
        </div>
      </div>
    </LayoutWrapperMain>
    <LayoutWrapperFooter
      className={null}
      rootClassName={null}
    >
      <injectIntl(Footer) />
    </LayoutWrapperFooter>
  </LayoutSingleColumn>
</Page>
`;
